### DISCUSSION TITLE
How to use `useNow` to display current time on a given timezone?

### DISCUSSION THREAD

**Discussion by stefanobartoletti** [vueuse/vueuse#3865] [Q&A] [answered]
I need to display the current time, but aspecific to a given timezone.

So in example, always show the current time of Central European Timezone, even when the user is visiting the site from let's say the Americas or Australia.

I've tried to read the docs of both [`useNow`](https://vueuse.org/core/useNow) and [`useDateFormat`](https://vueuse.org/shared/useDateFormat/), but I don't understand how to set the timezone, or if it is even possible by just using these composables alone.

Does somebody have suggestions about this? Thanks!

**Comment by daiwanxing** [comment, marked as answer]
that's easy,  although `useNow` doesn't provide timezone option,however, we can use Javascript built-in date api do it.

let me show you the below code:

```js
const { now } = useNow({ controls: true })

const userTimeZone = Intl.DateTimeFormat().resolvedOptions().timeZone;

const dateTimeFormatCache = {};

function formatTimeInTimeZone(date, timeZone, locale = 'en-US') {
   if (!dateTimeFormatCache[timeZone]) {
        dateTimeFormatCache[timeZone] = new Intl.DateTimeFormat(locale, {
            timeZone,
            year: 'numeric',
            month: '2-digit',
            day: '2-digit',
            hour: '2-digit',
            minute: '2-digit',
            second: '2-digit',
            hour12: true
        });
    }
    return dateTimeFormatCache[timeZone].format(date);
}

const timeInUserTimeZone= computed(() => formatTimeInTimeZone(now.value, userTimeZone));

// no matter which country users visit your website, they will always get the correct time zone information.
console.log(`Time in user's time zone (${userTimeZone}): ${timeInUserTimeZone.value}`);
```
If you never use the `Intl.DateTimeFormat` API, you can read some doc about this part on MDN.
